lib/client/client.go

```diff
@@ -1571,7 +1653,31 @@ func (c *NodeClient) RunCommand(ctx context.Context, command []string) error {
 	)
 	defer span.End()
 
-	nodeSession, err := newSession(ctx, c, nil, c.TC.newSessionEnv(), c.TC.Stdin, c.TC.Stdout, c.TC.Stderr, c.TC.EnableEscapeSequences)
+	// Set up output streams
+	stdout := c.TC.Stdout
+	stderr := c.TC.Stderr
+	if c.hostname != "" {
+		stdout = newLineLabeledWriter(c.TC.Stdout, c.hostname)
+		stderr = newLineLabeledWriter(c.TC.Stderr, c.hostname)
+
+		if c.sshLogDir != "" {
+			stdoutFile, err := os.Create(path.Join(c.sshLogDir, c.hostname+".stdout"))
```

Comment:
```suggestion
			stdoutFile, err := os.Create(filepath.Join(c.sshLogDir, c.hostname+".stdout"))
```

More here and below.